pkg/servicenameresolutiondetector/store/node_store_test.go

```diff
@@ -0,0 +1,332 @@
+				_, err = clientset.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+				assert.NoError(t, err)
+			}
+
+			var result interface{}
+			var opErr error
+
+			switch tc.operation {
+			case "Load":
+				result, opErr = store.Load(tc.key)
+			case "Store":
+				opErr = store.Store(tc.key, tc.inputCondition)
+			case "ListAll":
+				result, opErr = store.ListAll()
+			}
+
+			if tc.expectedError {
+				assert.Error(t, opErr)
+			} else {
+				assert.NoError(t, opErr)
+				if tc.expectedResult != nil {
+					assert.Equal(t, tc.expectedResult, result)
+				}
+			}
+		})
+	}
+}
+
+func TestNodeStore_ListAll(t *testing.T) {
+	clientset := fake.NewSimpleClientset()
+	nodeInformer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{},
+		&corev1.Node{},
+		0,
+		cache.Indexers{},
+	)
+	nodeLister := listcorev1.NewNodeLister(nodeInformer.GetIndexer())
+
+	store := NewNodeConditionStore(
+		clientset.CoreV1().Nodes(),
+		nodeLister,
+		string(corev1.NodeReady),
+		5*time.Minute,
+	)
+
+	testCases := []struct {
+		name           string
+		nodes          []*corev1.Node
+		expectedResult []metav1.Condition
+		expectedError  bool
+	}{
+		{
+			name: "List all ready nodes",
+			nodes: []*corev1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node2"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+			},
+			expectedResult: []metav1.Condition{
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionTrue},
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionTrue},
+			},
+			expectedError: false,
+		},
+		{
+			name: "List with not ready and stale nodes",
+			nodes: []*corev1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node2"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionFalse, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node3"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.NewTime(time.Now().Add(-10 * time.Minute))},
+						},
+					},
+				},
+			},
+			expectedResult: []metav1.Condition{
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionTrue},
```

Comment:
`listAll` function does not guarantee that the returned result will maintain the same order as the input `nodes`, which can lead to intermittent test failures.

```shell
=== RUN   TestNodeStore_ListAll/List_with_not_ready_and_stale_nodes
    node_store_test.go:249: 
                Error Trace:    /root/home/workspace/code/karmada/pkg/servicenameresolutiondetector/store/node_store_test.go:249
                Error:          Not equal: 
                                expected: "True"
                                actual  : "Unknown"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -(v1.ConditionStatus) (len=4) "True"
                                +(v1.ConditionStatus) (len=7) "Unknown"
                                 
                Test:           TestNodeStore_ListAll/List_with_not_ready_and_stale_nodes
    node_store_test.go:249: 
                Error Trace:    /root/home/workspace/code/karmada/pkg/servicenameresolutiondetector/store/node_store_test.go:249
                Error:          Not equal: 
                                expected: "Unknown"
                                actual  : "True"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -(v1.ConditionStatus) (len=7) "Unknown"
                                +(v1.ConditionStatus) (len=4) "True"
                                 
                Test:           TestNodeStore_ListAll/List_with_not_ready_and_stale_nodes
```

Reply:
I can sort the results before comparing.

Reply:
Looks like you are converting the result array into a map to avoid ordering issues, which makes sense to me

Reply:
Can we move ahead with it?

Reply:
yes, please go ahead

Reply:
@zhzhuang-zju can you trigger a retest?